src/SampleLinter/Rules/NoGroupsLinterRule.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Linting.Interfaces;
+using Dynamo.Linting.Rules;
+
+namespace SampleLinter.Rules
+{
+    public class NoGroupsLinterRule : GraphLinterRule
+    {
+        public override string Id => "EF17A093-4E94-4AE9-8876-18631396A23A"; //this id is unique to each linter rule
+        public override SeverityCodesEnum SeverityCode => SeverityCodesEnum.Warning;
+        public override string Description => $"There are {_nodesNotInGroups.Count} nodes that are not in groups.";
+        public override string CallToAction => $"You have nodes that are not in groups. Grouping your nodes is a best-practice when authoring Dynamo graphs. {_ungroupedAllowed} nodes are allowed outside of groups based on the Linter Settings.";
```

Comment:
I see that the other samples are localized so I think we'll need to do the same.

Reply:
Hmm, have some guidance for that?

Reply:
Yes, checkout Properties.Resources.LocalStringResult definition in SampleLibraryUI project